Refont du site, avec les idées du site de codealuxz

liankorExe opened the pull request:
Après avoir vu le code de Codealuxz, j'ai décidé de reprendre la codebase du site initial de ServerOpenMc et de le refaire moi-même proprement, avec de bons composants et un code clean.

liankorExe committed:
feat(Navbar): Implement responsive sheet menu and improve scroll behavior

- Added a responsive sheet menu using Radix UI for mobile navigation.
- Enhanced scroll behavior to adjust navbar height and style based on scroll position.
- Integrated overflow control for body when the menu is open.
- Updated navbar links rendering and button interactions.
- Added loading and error states for contributors section.
- Created Contributors component to fetch and display GitHub contributors for Website and PluginV2.
- Introduced Sheet component for consistent modal behavior across the application.
- Updated package dependencies to include @radix-ui/react-dialog.

liankorExe committed:
fix: Correct French contractions in JoinPage and Navbar components

liankorExe committed:
chore: Remove unnecessary blank line in JoinPage component

liankorExe committed:
fix: Update dark mode background color in globals.css

iambibi commented:
possible d'avoir un exemple du site?

AxenoDev commented:
un, gh page

liankorExe commented:
> possible d'avoir un exemple du site?

yes, j'essaye de mettre en place un workflow là mais j'ai un petit soucis je fais au plus vite

liankorExe commented:
> un, gh page

check discord dans tes mps stp

liankorExe committed:
fix: Update card background color in dark mode and adjust JoinPage styling

liankorExe committed:
gsfd

liankorExe committed:
Update nextjs.yml

liankorExe committed:
chore: Remove GitHub Actions workflow for Next.js deployment

liankorExe committed:
Merge branch 'main' of https://github.com/liankorExe/Website

liankorExe committed:
workflow deleted

liankorExe committed:
Create nextjs.yml

liankorExe committed:
mettre à jour la couleur d'arrière-plan du thème sombre et améliorer l'interface utilisateur de JoinPage avec le composant Carte de Shadcn. Utilisation des boutons shadcn + nouvelle variante Discord.

liankorExe committed:
Merge branch 'main' of https://github.com/liankorExe/Website

liankorExe committed:
Titre contributeurs changé

liankorExe commented:
> possible d'avoir un exemple du site?

Voilà le resultat : https://liankorexe.github.io/Website/

liankorExe committed:
Titre changé lors du loading

AxenoDev commented on app/globals.css:
```diff
@@ -80,9 +80,9 @@
-  --background: oklch(0.145 0 0);
+  --background: #12090c;
   --foreground: oklch(0.985 0 0);
-  --card: oklch(0.205 0 0);
```
laisse les couleurs de base

liankorExe committed:
Ajout d'un composant de footer avec système d'affichage et de mise en cache des contributeurs GitHub

- Nouveau Changelog
- Implémentation d'un nouveau composant footer qui récupère et affiche les contributeurs du dépôt GitHub « ServerOpenMC/Website ».
- Introduction d'un mécanisme de mise en cache pour les appels d'API GitHub utilisant localStorage afin d'optimiser les performances et de réduire les limites de débit des API.
- Création de composants d'interface utilisateur réutilisables pour Avatar, Badge, Dialogue, ScrollArea, Séparateur et Tooltip.
- Amélioration de l'expérience utilisateur avec des états de chargement et des tooltips pour les informations des contributeurs.

liankorExe committed:
Ajout de la prise en charge des images GitHub dans la configuration Next.js et amélioration des types pour les contributeurs et les statistiques dans plusieurs composants.

liankorExe committed:
fix avatar barre noire au hover

EstePrime commented on components/home/header.tsx:
Tous les éléments de la page d’accueil sont ici. Tu devrais les découper en plusieurs parties, puis les importer dans la page principale `page.tsx` + le chargement des contributeurs empêche toute interaction avant qu’il soit terminé, ce qui donne l’impression que la page est buggée. Sépare mieux tes composants et privilégie le rendu côté serveur, ou encore mieux (si possible), le rendu statique au moment du build.

SSR : https://nextjs.org/docs/pages/building-your-application/rendering/server-side-rendering
SSG : https://nextjs.org/docs/pages/building-your-application/rendering/static-site-generation

https://github.com/user-attachments/assets/86e0ba76-4bbd-453e-8387-3b57772df022

EstePrime commented on components/home/header.tsx:
```diff
@@ -7,103 +7,251 @@ import Link from "next/link";
+                  className="flex items-center"
+                >
+                  Contribuer
+                  <svg
```
Essaie d'importer les icônes depuis un autre fichier genre icons.tsx, ou alors utilise directement ceux de la lib déjà présente (`lucide-react`).

EstePrime replied:
Je pointais le SVG, mais GitHub en a décidé autrement 😑

liankorExe replied:
fait

EstePrime commented on next.config.ts:
```diff
@@ -1,7 +1,22 @@
+      },
+    ],
+  },
 };
```
Le `avatars.githubusercontent.com` est inutile puisque que `*.githubusercontent.com` englobe déjà tous les sous-domaines.

```ts
const nextConfig: NextConfig = {
  images: {
    remotePatterns: [
      {
        protocol: "https",
        hostname: "*.githubusercontent.com",
        port: "",
        pathname: "/**",
      },
    ],
  },
};
```

EstePrime replied:
Je pointais le remotePatterns, mais GitHub en a décidé autrement 😑

liankorExe replied:
good

EstePrime reviewed (changes requested):
Bonne utilisation de Tailwind et des composants de Shadcn. C’est une implémentation bien plus soignée que celle de Codelaluxz. Une vraie compréhension de la codebase et pas simplement Github Copilot ou Cursor lancé au hasard.

AxenoDev commented on app/globals.css:
```diff
@@ -80,7 +80,7 @@
 }
 
 .dark {
-  --background: oklch(0.145 0 0);
```
nan mais laisse le bg de base

liankorExe replied:
fait

EstePrime commented on components/navigation/navbar.tsx:
```diff
@@ -3,138 +3,149 @@
+    <>
+      <header
+        className={`fixed top-0 left-0 w-full z-[9999] flex justify-center items-center transition-all duration-300 ease-in-out
+                  ${isScrolled ? "h-[130px]" : "h-[70px]"}`}
```
Enlève le `h-[70px]`, ça crée un écart inégal entre le haut et le bas de la navbar.

Avec :
![image](https://github.com/user-attachments/assets/299d4031-59ca-4500-84cd-f757b4529873)

Sans : 
![image](https://github.com/user-attachments/assets/0d365975-f117-4a86-a9f8-94c57a262633)

liankorExe replied:
si j'enleve le h-[70px] la barre va se decaler du top
![image](https://github.com/user-attachments/assets/ea410950-db1a-487d-ae64-54d134ebbc2d)

liankorExe replied:
j'ai mis 80 du coup
![image](https://github.com/user-attachments/assets/13055105-8ae0-4900-b9c2-b35dd5a8f4c4)

liankorExe committed:
Refactorisation du composant Header et division en plusieurs composants

- Suppression de la logique complexe et des animations du composant Header.
- Introduction des composants HeroSection, HeroImageWithStats et ContributorsSection.
- Implémentation du chargement différé pour la section Contributeurs.
- Ajout du composant GitHubStats pour récupérer et afficher les statistiques GitHub.
- Mise à jour de la hauteur de la barre de navigation lors du défilement.
- Suppression des journaux de la console des méthodes de cache GitHubApi.
- Ajout de la dépendance @radix-ui/react-hover-card pour la fonctionnalité de carte de survol.

liankorExe committed:
Refactor Navbar component by removing unnecessary comments and improving code readability

liankorExe committed:
Utilisation de l'icon Heart à la place d'un svg

AxenoDev commented on app/join/page.tsx:
```diff
@@ -1,60 +1,405 @@
+        </div>
+      </main>
+
+      <footer className="border-t bg-muted/50 mt-20">
```
Enlève le footer

AxenoDev replied:
car sinon il y en a deux

liankorExe replied:
fait

AxenoDev commented on components/layout/footer.tsx:
```diff
@@ -0,0 +1,332 @@
+            <div className="space-y-4">
+              <div className="flex items-center gap-2">
+                <div className="w-8 h-8 bg-primary rounded-lg flex items-center justify-center">
+                  <span className="text-primary-foreground font-bold text-sm">
```
mets le logo d'OMC

liankorExe replied:
fait

AxenoDev replied:
![image](https://github.com/user-attachments/assets/20f5e197-e6b5-40f1-9f33-16f6c63e99b0)
hmm, cassé un peu le logo

liankorExe replied:
où ça ? parce que j'ai pas de probleme perso meme en navigation privée sans cache

liankorExe replied:
ah si pardon
je regarde ça

liankorExe committed:
suppression du deuxieme footer dans la page join + ajout du logo OMC dans le footer

liankorExe committed:
Page galerie

liankorExe committed:
Logo openMc footer fix

iambibi commented:
@liankorExe pk c'est en draft ? c'est pas pret?

AxenoDev commented:
l'icon dans le footer n'est pas bon 
![image](https://github.com/user-attachments/assets/0f44e5e8-9a08-4c8f-8cc0-561efdf96d41)

AxenoDev commented on components/navigation/navbar.tsx:
```diff
@@ -3,138 +3,170 @@
+                      }
+                      ${
+                        isScrolled
+                          ? "w-[calc(100%-60px)] max-w-[1300px] mx-[30px] rounded-[30px] px-6 py-4"
```
réduis le rounded, et fait un autre rounded pour le responsive

liankorExe replied:
j'ai mis un rounded de 20px je te laisse aller voir et me dire si c'est bien, et pour un rounded pour un autre rounded pour le responsive tu entends quoi par là ?

AxenoDev replied:
genre sur téléphone il est trop fort, donc met plus du `rounded-md` ou du style

AxenoDev commented on components/home/_components/contributors.tsx:
```diff
@@ -0,0 +1,337 @@
+          rel="noopener noreferrer"
+          className="text-primary hover:text-primary/80 transition-colors font-medium text-sm"
+        >
+          Voir tous les contributeurs sur GitHub →
```
mets une fleche avec `lucide-react`

liankorExe replied:
fait

AxenoDev commented on components/home/_components/contributors.tsx:
```diff
@@ -0,0 +1,337 @@
+              </p>
+              {(contributor.additions || contributor.deletions) && (
+                <p className="text-xs text-muted-foreground">
+                  {contributor.netLines !== undefined &&
```
Je ne sais plus exactement comment on fait, mais il y a un truc du style
```ts
{contributor.netLines !== undefined && contributor.netLines >= 0 && "+"}

liankorExe replied:
AH oui c'est bon j'ai modif

AxenoDev commented on app/gallery/page.tsx:
```diff
@@ -0,0 +1,409 @@
+  participants?: number;
+}
+
+const galleryImages: GalleryImage[] = [
```
![image](https://github.com/user-attachments/assets/1267a6fd-16aa-413a-b5f9-8be64cd59599)

AxenoDev replied:
Il ne trouve pas les images à load

liankorExe replied:
c'est fix, il y avait un probleme avec l'url du site du coup

liankorExe committed:
Update page.tsx

liankorExe committed:
fix footer logo

liankorExe commented:
> l'icon dans le footer n'est pas bon ![image](https://private-user-images.githubusercontent.com/93048526/463795653-0f44e5e8-9a08-4c8f-8cc0-561efdf96d41.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTIwNzExNTEsIm5iZiI6MTc1MjA3MDg1MSwicGF0aCI6Ii85MzA0ODUyNi80NjM3OTU2NTMtMGY0NGU1ZTgtOWEwOC00YzhmLThjYzAtNTYxZWZkZjk2ZDQxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzA5VDE0MjA1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRkN2I4ZmE5Y2E2NTg5ODRlYzA1NzAzODlmMmMxZWM4YjFiOTY0MzM0ZDdiYTRmOTU4MzczMjFkNjhkZGEyOTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7d1IhvHbAX0Ju5ovS3HfJBaMdRdal__FYKi51RG5prc)

fix

liankorExe committed:
rounded moins fort 30px -> 20px

liankorExe committed:
Update contributors.tsx

remplacement de la fleche par un chevron avec lucide + clean du code au niveau des netLines

liankorExe committed:
Update page.tsx

fix images qui se loadaient pas sur le workflow car l'url est en /Website mais en local il n'y est pas

liankorExe committed:
Update page.tsx

liankorExe committed:
Update page.tsx

meilleur bordures pour les tags dans galerie

liankorExe committed:
Update page.tsx

fix du s à jour quand il y en avait qu'un seul

liankorExe committed:
fix details changelog en dessous de la navbar

liankorExe committed:
fix navbar qui s'affichaient dans le hamburger menu

AxenoDev commented:
https://github.com/user-attachments/assets/6eeb06e4-d486-4ed4-b832-bfb9dca4f2c6

Petit problème avec l'animation dans Galerie

liankorExe committed:
fix animation avec le hover

liankorExe commented:
> 2025-07-15_17-38-40.mp4 
> Petit problème avec l'animation dans Galerie

c'est fix

AxenoDev reviewed (changes requested):
<img width="288" height="62" alt="image" src="https://github.com/user-attachments/assets/7408128a-0dec-4690-b04f-41c4b77e4fe5" />
Augmente la taille du btn

liankorExe committed:
bouton changelog complet plus grand en hauteur

liankorExe commented:
> <img alt="image" width="288" height="62" src="https://private-user-images.githubusercontent.com/93048526/471038580-7408128a-0dec-4690-b04f-41c4b77e4fe5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM1MzU0MjMsIm5iZiI6MTc1MzUzNTEyMywicGF0aCI6Ii85MzA0ODUyNi80NzEwMzg1ODAtNzQwODEyOGEtMGRlYy00NjkwLWIwNGYtNDFjNGI3N2U0ZmU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzI2VDEzMDUyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg0MmEyOTgwNzMxYjQzYzU5MzVkMDEyYmE3NGU1YzQxNGYyM2RmNzU2N2I0OTczYTAxZmI5ZTcyYjM4Zjk1ZDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VrdvaYCc0tfaP_ZWfV3fyI5URH3T9kPbc82hUvRxymg"> Augmente la taille du btn

c'est good

Margouta committed:
Merge branch 'main' into main

Margouta commented:
Ok pour moi après test sur plusieurs devices
